hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/grid/AbstractGrid.java

```diff
@@ -8,40 +8,70 @@
  */
 package org.eclipse.hawkbit.ui.common.grid;
 
+import java.util.Arrays;
+import java.util.Locale;
+
 import org.eclipse.hawkbit.ui.SpPermissionChecker;
 import org.eclipse.hawkbit.ui.components.RefreshableContainer;
-import org.eclipse.hawkbit.ui.utils.VaadinMessageSource;
+import org.eclipse.hawkbit.ui.utils.SPDateTimeUtil;
 import org.eclipse.hawkbit.ui.utils.SPUIDefinitions;
+import org.eclipse.hawkbit.ui.utils.VaadinMessageSource;
 import org.vaadin.addons.lazyquerycontainer.LazyQueryContainer;
 import org.vaadin.spring.events.EventBus;
 import org.vaadin.spring.events.EventBus.UIEventBus;
 
-import com.vaadin.data.Container;
 import com.vaadin.data.Container.Indexed;
+import com.vaadin.data.util.GeneratedPropertyContainer;
+import com.vaadin.data.util.converter.Converter;
 import com.vaadin.ui.Grid;
 
 /**
- * Abstract table class.
+ * Abstract grid that offers various capabilities (aka support) to offer
+ * convenient enhancements to the vaadin standard grid.
  *
+ * @param <T>
+ *            The container-type used by the grid
  */
-public abstract class AbstractGrid extends Grid implements RefreshableContainer {
-
+public abstract class AbstractGrid<T extends Indexed> extends Grid implements RefreshableContainer {
     private static final long serialVersionUID = 4856562746502217630L;
 
     protected final VaadinMessageSource i18n;
-
     protected final transient EventBus.UIEventBus eventBus;
-
     protected final SpPermissionChecker permissionChecker;
 
-    protected AbstractGrid(final VaadinMessageSource i18n, final UIEventBus eventBus, final SpPermissionChecker permissionChecker) {
+    protected transient AbstractMaximizeSupport maximizeSupport;
```

Comment:
There are a lot of reasons why you should avoid protected member variables:

http://softwareengineering.stackexchange.com/questions/162643/why-is-clean-code-suggesting-avoiding-protected-variables